src/preproc/preprocessing_pass_api.h

```diff
@@ -0,0 +1,46 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_api.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Preprocessing pass API for passes
+ **
+ ** The preprocessing pass API is the interface between solver and
+ ** preprocessing passes. Passes are expected to use API for 
+ ** read only access.
+ **/
+#include "cvc4_private.h"
+
+#ifndef __CVC4__PREPROC__PREPROCESSING_PASS_API_H
+#define __CVC4__PREPROC__PREPROCESSING_PASS_API_H
+
+#include <string>
+#include "decision/decision_engine.h"
+#include "smt/smt_engine.h"
+#include "theory/arith/pseudoboolean_proc.h"
+#include "theory/booleans/circuit_propagator.h"
+#include "theory/theory_engine.h"
+
+namespace CVC4 {
+namespace preproc {
+
+class PreprocessingPassAPI {
```

Comment:
API does not add any information. Please rename the class and files.

Reply:
Tim - the API will get filled in as the actual passes come in.  That said, I thought we agreed to start with the engines in the API.

Reply:
Yes, I think that you are right, @barrettcw . I also agree that PreprocessingPassContext is a better name.